community/contributing/pr_review_guidelines.rst

```diff
@@ -0,0 +1,335 @@
+
+Remember, that a good code review is not only beneficial for the
+codebase itself, but also for the reviewee and the reviewer too. You
+make better contributors out of PR authors when you give them a good
+thoughtful review, or even when you just offer small tips. And you
+can learn something new yourself.
+
+While a proper code review requires you to understand the nuance of the
+affected engine area and is better left for your judgement, there is a
+checklist of universal things to look for:
+
+* **The PR follows** :ref:`doc_code_style_guidelines`.
+
+  While ``clang-format`` and various CI checks can catch a lot of
+  inconsistencies, they are far from perfect in that regard and are
+  unable to detect some issues. For example, check that:
+
+  * The style of header includes is respected;
+  * Identifiers use ``snake_case`` and follow our naming conventions;
+  * Method parameters start with ``p_*`` or ``r_*``.
+
+* **Code only touches the areas announced in the PR (and the commit
+  message).**
+
+  It is always tempting to start fixing random things in code, as you
+  see them. However, this can quickly become a hell to dig through in
+  the commit history. Small touch-ups next to the related area are
+  alright, but often bugs that you can find along the way are better
+  left for their own PRs.
+
+* **Code properly uses Godot's own APIs and patterns.**
+
+  This is true for any reasonably sized project — consistency is very
+  important, and a solution that already exists in the codebase
+  is preferable to an ad-hoc solution.
+
+* **Are core areas affected by the change?**
+
+  Sometimes a PR that is supposed to solve a local problem can have a
+  far-reaching effect, way outside of its scope. If you believe that
+  is the case, make sure to get senior maintainers involved in the
+  process, namely `reduz <https://github.com/reduz>`_ or
+  `vnen <https://github.com/vnen>`_. Their input and approval would
+  be required for changes to the engine's core systems and overall
+  architecture, APIs that touch on all Objects or Nodes, changes
+  to GDScript and GDNative. If the pull request modifies the build
+  system, you can contact `Akien <https://github.com/akien-mga>`_.
+
+5. Iterate with the contributor and improve the PR
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Inevitably, a lot of pull requests would be initially undercooked.
+At this point, you and the contributor should enter a feedback loop
+where they iterate on your notes, recommendations and requests, and
+you look that they have indeed fixed the issues you've highlighted.
+
+Try not to exhaust the contributor with style nitpicks, especially
+if it's still up in the air whether their PR would be accepted at
+all. Preferably, suggestions should come in order of importance:
+first address their overall code design and approach to solving the
+problem, then make sure their code is complying with the engine's
+best practices, and last do the "pretty pass".
```

Comment:
The problem that I have with this suggestion is that the set of people who can make a decision about whether a PR is taking the right approach is different from and much smaller than the set of people who can give reviews with simple things such as style nitpicks. In such a case it's not clear whether to give small reviews or do nothing.

Reply:
This is derived from the note from @akien-mga that we shouldn't exhaust contributors with code style nitpicks. Especially not before the feature overall has been approved. I myself sometimes open an issue a leave a comment about something mundane because I notice it, but don't do a proper review overall. This may actually hurt the process, as others may expect me to continue with a proper review, and the contributor may think that they don't have other problems since I am paying attention to the small stuff.

In the end, I don't think there is a clear "do" or "don't" here. The word "Preferably" points to it being a suggestion and not a rule too.